git2-curl/tests/all.rs

```diff
@@ -15,6 +15,7 @@ const PORT: u16 = 7848;
 
 fn main() {
     unsafe {
+        libgit2_sys::init();
```

Comment:
This seems somewhat suspicious in the sense that `git2_curl` should internally not require this but instead do it on behalf of libraries. I think that there's perhaps a missing call to initialization somewhere inside of `git2` itself, since `git2_curl` only uses the APIs of `git2`?

Reply:
Yea, I meant that `init` should be called *from* `git2_curl::register`, not with.

Reply:
I moved the init call into `git2_curl::register`. 69334f8
Feels like I'm missing a way to get this init'd without adding a direct dependency on libgit2_sys, but I'm not seeing it. (noob alert).